data/shell-completions/zsh/meson-comp-helper.py

```diff
@@ -0,0 +1,225 @@
+#!/usr/bin/python3
+
+'''This file generates completion function definitions for Zsh.'''
+# NB: the function definitions are directly passed to the `eval` builtin.
+# We must take care not to trust meson introspect's output
+# and prevent e. g. code injection.
+
+import json
+import os
+import sys
+from shlex import quote
+
+
+def fsemic(s):
+    return s.replace(':', r'\:').replace(r'\\:', r'\:')
+
+
+def dsemic(s):
+    return s.replace(':', r'\\:')
+
+
+def c_style_octal(i):
+    return '0{:03}'.format(int(oct(i)[2:]))
+
+
+def generate_choices(optentry):
+    '''If a build option can be set to a limited number of values,
+       return an iterable of possible values. Otherwise, return None.
+    '''
+    if optentry['type'] in ('combo', ):
+        return optentry.get('choices', [])
+    if optentry['type'] in ('boolean', ):
+        return ('false', 'true')
+    return None
+
+
+def simple_hint(i):
+    return '<{}>'.format(i)
+
+
+def generate_hint(optentry):
+    '''Produce a string representation of a build option's current value.
+    '''
+    v = optentry['value']
+
+    if optentry['type'] in ('boolean', ):
+        return simple_hint('true' if v else 'false')
+    if optentry['name'] == 'install_umask':
+        return simple_hint(c_style_octal(v))
+    if optentry['type'] in ('integer', ):
+        return simple_hint(int(v))
+
+    return simple_hint(repr(v))
+
+
+def zsh_tag(k):
+
+    def asciify(c):
+        return c if ord(c) < 0x80 and c.isalnum() else '-'
+
+    return ''.join(map(asciify, k))
```

Comment:
This function can be reduced to:
```python
return ''.join(c if ord(c) < 0x80 and c.isalnum() else '-' for c in k)
```
`map` is kinda deprecated and we try to avoid using it in meson.

Reply:
OK, no objections